File: y3/sem_b/android/ChattyGroup/app/src/main/java/com/yali_noam/chattygroup/SplashActivity.java
package com.yali_noam.chattygroup;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.view.animation.OvershootInterpolator;
import android.widget.ImageView;

import com.google.firebase.auth.FirebaseAuth;

public class SplashActivity extends AppCompatActivity {

    int animationTime = 4000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        ImageView logo = findViewById(R.id.logo);
        logo.animate()
                .rotationYBy(360*3)
                .setDuration(animationTime)
                .setInterpolator(new OvershootInterpolator())
                .start();

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if(FirebaseAuth.getInstance().getCurrentUser() == null){
                    startActivity(new Intent(SplashActivity.this, LoginActivity.class));
                    finish();
                }
                else {
                    startActivity(new Intent(SplashActivity.this, ChatPartnersActivity.class));
                    finish();
                }
            }
        }, animationTime+1500);
    }
}

File: y3/sem_b/android/chatty_betty/app/src/main/java/com/android/noam/chatty_betty/SplashActivity.kt
package com.android.noam.chatty_betty

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.OvershootInterpolator
import android.widget.ImageView
import android.content.Intent
import android.os.Handler
import com.google.firebase.auth.FirebaseAuth




class SplashActivity : AppCompatActivity() {

    var animationTime = 4000

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)


        val logo : ImageView = findViewById(R.id.logo)
        logo.animate()
                .rotationYBy((360*3).toFloat())
                .setDuration(animationTime.toLong())
                .setInterpolator(OvershootInterpolator())
                .start()

        Handler().postDelayed({
            if (FirebaseAuth.getInstance().currentUser == null) {
                startActivity(Intent(this@SplashActivity, LoginActivity::class.java))
                finish()
            } else {
                startActivity(Intent(this@SplashActivity, ChatPartnersActivity::class.java))
                finish()
            }
        }, (animationTime + 1500).toLong())

    }
}
